scenes/test_menu.txt

```diff
@@ -0,0 +1,173 @@
+*label entry
+*params module log_buffer
+*temp test_count 0
+*temp tests_passed 0
+
+*comment Override choice UI with an automated system:
+*comment - cslib_choice_select: A list of of options to select sequentially
+*comment - cslib_choice_count: The number of 'visible' options at the end of the sequence
+*comment - cslib_choice_n: The 'visible' options at the end of the sequence
+*comment - cslib_ret: The last 'selected' option at the end of the sequence
+*set cslib_catch_choice true
+
+*comment :::::: SIMPLE tests ::::::
+*gosub test_start "simple - single item"
+*set cslib_choice_select "1"
+*gosub_scene cslib_menu build_simple "Hello"
+*gosub test_assert_equal 1 cslib_choice_count
+*gosub test_assert_equal 1 cslib_ret
+*gosub test_assert_equal "Hello" cslib_choice_1
+*gosub test_finish
+
+*gosub test_start "page-count invalid"
+*temp result_expected "Bug: cslib_menu._render_choice: this option should not be selectable"
+*set cslib_choice_select "1"
+*set cslib_catch_bug true
+*gosub_scene cslib_menu build_simple 1 2 3 4 5 6
+*set cslib_catch_bug false
+*gosub test_finish
+
+*gosub test_start "simple - two items (with cancel)"
+*set cslib_choice_select "3"
+*gosub_scene cslib_menu build_simple_with_cancel "Hello" "Goodbye"
+*gosub test_assert_equal 3 cslib_choice_count
+*gosub test_assert_equal 0 cslib_ret
+*gosub test_assert_equal "Hello" cslib_choice_1
+*gosub test_assert_equal "Goodbye" cslib_choice_2
+*gosub test_assert_equal "[return]" cslib_choice_3
+*gosub test_finish
+
+*gosub test_start "simple - single-page (page 1, option 1)"
+*set cslib_choice_select "3"
+*gosub_scene cslib_menu build_simple 1 2 3 4 5 6 7 8 9 10 11
+*gosub test_assert_equal 7 cslib_choice_count
+*gosub test_assert_equal 1 cslib_ret
+*gosub test_assert_equal "Page (1 / 3)" cslib_choice_1
+*gosub test_assert_equal "[i]Next -->[/i]" cslib_choice_2
+*gosub test_assert_equal "1" cslib_choice_3
+*gosub test_assert_equal "2" cslib_choice_4
+*gosub test_assert_equal "3" cslib_choice_5
+*gosub test_assert_equal "4" cslib_choice_6
+*gosub test_assert_equal "5" cslib_choice_7
+*gosub test_finish
+
+*gosub test_start "simple - multi-page (navigation)"
+*comment Select 'Next' then 'Next' then 'Previous', then '8' on Page 2
+*set cslib_choice_select "2,2,2,5"
```

Comment:
oooh, you need multiple choices for pagination, right?

Reply:
Correct.